app/models/activity.rb

```diff
@@ -0,0 +1,62 @@
+# Activity model inherited from public_activity gem is used to customize activities' management
+class Activity < PublicActivity::Activity
+  acts_as_readable on: :created_at
+  after_create :send_email
+
+  # The formats of notification that are supported
+  SUPPORTED_FORMATS = [:feed, :email, :popup]
+
+  # Class methods used to create an activity
+  class << self
+    # Create an activity record and store the notification formats of it
+    #
+    # @param from the one triggers this activity
+    # @param [User] to the user who will receive notifications
+    # @param trackable the object that will be tracked in this activity
+    # @param [Symbol] event the name of the activity indicating which event happened
+    # @param [Array<Symbol>] formats the formats of the notifications that will be used to inform
+    #   users. Any combination of :activity_feed, :email and :popup, is valid
+    def create(from, to, trackable, event, formats: [])
+      activity_params = { owner: from, recipient: to, trackable: trackable }
+      activity_params[:key] = generate_key(trackable, event)
+      formats.each do |format|
+        activity_params[format] = true if SUPPORTED_FORMATS.include?(format)
+      end
+      super(activity_params)
+    end
+
+    private
+
+    # Generate the key of activity
+    #
+    # @example generate_key(@course,:created) => courses.activities.created
+    #
+    # @param trackable The object that will be tracked
+    # @param [Symbol] event the name of the activity indicating which event happened
+    # @return [String]
+    def generate_key(trackable, event)
+      "#{trackable.class.name.underscore.pluralize}.activities.#{event}"
+    end
+  end
+
+  # Translate the key of an activity to the path of a given activity format
+  #
+  # @example template_path(:popup) => /ClassName/activities/Event/popup
+  #
+  # @param [Symbol] notification_format the format of notification
+  # @return [String] the path of a template
+  def template_path(notification_format)
+    return unless SUPPORTED_FORMATS.include?(notification_format)
+    path = key.split('.')
+    format("/%s/#{notification_format}", path.join('/'))
```

Comment:
You can express the destination string in fewer words. You are just doing a string replace and appending the notification format. Why do you need to split, join, then concatenate using a format string?